### PR TITLE
Code cleanups from PVS/Studio

### PR DESCRIPTION
**Pull request by greglandrum**
The team that makes PVS/Studio makes free licenses available for open-source projects (which is an awesome service to the community).

I ran it across the RDKit and have started fixing things that it flags up. This PR is the first wave of those fixes. I haven't made it through the report completely yet, but I will do another round at some point in the hopefully not-too-distant future.

Most of this is at the level of theoretical problems or removing warnings, but there are a couple of actual obvious bugs in there that either haven't come up yet or no one has noticed. I'll add comments for those.

### CODE REVIEW

File: Code/Geometry/point.cpp [hunk 1]
```diff
@@ -116,7 +116,7 @@ PointND operator+(const PointND& p1, const PointND& p2) {
   if (p1.dimension() < p2.dimension()) {
     dim = p1.dimension();
   } else {
-    dim = p1.dimension();
+    dim = p2.dimension();
```
Comment (greglandrum) [inline Code/Geometry/point.cpp]: This (and the same change at line 132) was a bug.

File: Code/GraphMol/AddHs.cpp [hunk 2]
```diff
@@ -396,7 +396,7 @@ void AssignHsResidueInfo(RWMol &mol) {
           h_label = h_label.substr(3, 1) + h_label.substr(0, 3);
           AtomPDBResidueInfo *newInfo = new AtomPDBResidueInfo(
               h_label, max_serial, "", info->getResidueName(),
-              info->getResidueNumber(), info->getChainId(), "",
+              info->getResidueNumber(), info->getChainId(), "", 1.0, 0.0,
```
Comment (greglandrum) [inline Code/GraphMol/AddHs.cpp]: This was a bug.

File: Code/GraphMol/FileParsers/ProximityBonds.cpp [hunk 3]
```diff
@@ -82,7 +82,7 @@ bool IsBlacklistedPair(Atom *beg_atom, Atom *end_atom) {
     if (IsBlacklistedAtom(beg_atom) || IsBlacklistedAtom(end_atom)) return true;
     // Dont make bonds to waters
     if (beg_info->getResidueName() == "HOH" ||
-        beg_info->getResidueName() == "HOH")
+        end_info->getResidueName() == "HOH")
```
Comment (greglandrum) [inline Code/GraphMol/FileParsers/ProximityBonds.cpp]: This was a bug

File: Code/Geometry/UniformGrid3D.cpp [hunk 4]
```diff
@@ -154,6 +154,7 @@ Point3D UniformGrid3D::getGridPointLoc(unsigned int pointId) const {
   }
   Point3D res;
   res.x = (pointId % d_numX) * d_spacing;
+  // the rounding here is intentional, we want the coordinates of a grid point
```
Comment (bp-kelley) [inline Code/Geometry/UniformGrid3D.cpp]: Can you remove the PVS warning by doing an explicit cast or an rdcast?

Comment (bp-kelley) [reply Code/Geometry/UniformGrid3D.cpp]:
```
res.x = static_cast<double>((pointId % d_numX)) * d_spacing;
res.y = statiic_cast<double>((pointId / (d_numX * d_numY))) * d_spacing;
```

Comment (bp-kelley) [reply Code/Geometry/UniformGrid3D.cpp]: Note that this is slightly different than your fix as we cast the entire integer operations to a double which is being done implicitly in the original code.

File: Code/GraphMol/ChemReactions/Reaction.h [hunk 5]
```diff
@@ -140,7 +140,7 @@ class RDKIT_CHEMREACTIONS_EXPORT ChemicalReaction : public RDProps {
       ROMol *agent = new ROMol(**iter);
       m_agentTemplates.push_back(ROMOL_SPTR(agent));
     }
-    dp_props = other.dp_props;
+    d_props = other.d_props;
```
Comment (bp-kelley) [inline Code/GraphMol/ChemReactions/Reaction.h]: Out of curiosity, why the change from dp_props to d_props?

Comment (greglandrum) [reply Code/GraphMol/ChemReactions/Reaction.h]: because it's not a pointer. At least I don't think it is

File: Code/Geometry/UniformGrid3D.cpp (hunk 4 above)
Comment (greglandrum) [reply Code/Geometry/UniformGrid3D.cpp]: The warning wasn't about a missing cast. It was about the fact that the operation as written was losing precision. That's actually by design (though I didn't realize that at first).

File: Code/GraphMol/ChemReactions/Reaction.h (hunk 5 above)
Comment (bp-kelley) [reply Code/GraphMol/ChemReactions/Reaction.h]: Correct :)

**Comment by bp-kelley** [comment]
@greglandrum I made a PR to your repo based on the FilterCatalog warnings, we could do this separately though.

**Comment by greglandrum** [comment]
> @greglandrum I made a PR to your repo based on the FilterCatalog warnings, we could do this separately though.

Thanks for pointing that out. I knew I had seen a notification about a PR from you about this, but then I couldn't find the PR. :-)
